File: Algorithms/JavaScript/invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 *
 * @param {TreeNode} root
 * @return {TreeNode}
 */

var invertBinaryTree = function(root) {
  if (root == null) return null;

  var temp = root.left;
  root.left = invertTree(root.right);
  root.right = invertTree(temp);
  
  return root;
};

File: Algorithms/Ruby/invert_binary_tree.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val)
#         @val = val
#         @left, @right = nil, nil
#     end
# end
#
# @param {TreeNode} root
# @return {TreeNode}

def invert_binary_tree(root)
  root.right, root.left = invert_binary_tree(root.left), invert_binary_tree(root.right) unless root.nil?
  root
end

